### PR TITLE
Feat: packet forward middleware

### PR DESCRIPTION
**Pull request by pr0n00gler**
**Merge with:** https://github.com/neutron-org/neutron-integration-tests/pull/112
**Tests run:** https://github.com/neutron-org/neutron-tests/actions/runs/4545383271

### CODE REVIEW

**Review by swelf19 — COMMENTED** [review]
What are the middleware sideeffects? does it charge some fees? What do you think to implement some unit tests?

File: proto/transfer/v1/tx.proto [hunk 1]
```diff
@@ -41,7 +41,9 @@ message MsgTransfer {
   uint64 timeout_timestamp = 7
       [ (gogoproto.moretags) = "yaml:\"timeout_timestamp\"" ];
 
-  neutron.feerefunder.Fee fee = 8
+  string memo = 8;
```
Comment (zavgorodnii) [inline proto/transfer/v1/tx.proto:44]: usually reordering of fields is not a good practice, is there any specific reason to do it here?

**Comment by zavgorodnii** [comment]
please add a documentation pr for the new module!

**Comment by zavgorodnii** [comment]
in the integration tests there is only a success scenario; probably we would also like to check that error propagation works properly. please implement n error scenario, or create a task to implement one

File: proto/transfer/v1/tx.proto (hunk 1 above)
Comment (mmikev) [reply proto/transfer/v1/tx.proto:44]: Yes, we want to save compatibility with the original transfer message from ibc-go

**Review by zavgorodnii — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

**Pull request merged**